magento/magento2
# SubTotal Rounding Issue For Decimal Quantity Enabled Product

### Description (*)
SubTotal Rounding Issue For Decimal Quantity Enabled Product

### Related Pull Requests
Insert Below Code

vendor/magento/module-directory/Model/Currency.php : Line No. 429

```
$this->numberFormatter->setAttribute(
            \NumberFormatter::ROUNDING_MODE,
            \NumberFormatter::ROUND_HALFUP
        );
```
### Fixed Issues (if relevant)
1. Fixes #37817 

### Manual testing scenarios (*)


::: Test Product Specification :::
-- Type : Simple
-- Price 20.67
-- Quantity : 1.5 (Enabled Decimal Quantity For Product)

::: Fixed Shipping Price :::
-- Price : 7.50 $

Steps to reproduce
Add a product, with above defined specifications, to cart. (Now look at checkout/cart page)

After that flat rate shipping price is set to 7.50 $, place order. And check customer account => Order History (You'll find incorrect order total, it was supposed to be 38.51 but showing 30.50) => View Order (Order Item Subtotal, Sub Total & Grand Total is wrong with value of 38.50, which is supposed to be 38.51).


### Contribution checklist (*)
